admin-dev/themes/new-theme/js/product-page/feature.js

```diff
@@ -0,0 +1,45 @@
+import $ from 'jquery';
+
+export default function() {
+  $(document).ready(function() {
+    var $featuresContainer = $('#features-content');
+
+    $featuresContainer.on('change', '.row select, .row input[type="text"]', (event) => {
+      var that = event.currentTarget;
+      var $row = $($(that).parents('.row')[0]);
+      var $featureSelector = $row.find('.feature-selector');
+      var $definedValueSelector = $row.find('.feature-value-selector');
+      var $customValueSelector = $row.find('input[type=text]');
+
+      // if feature has changed we need to reset values
+      if ($(that).hasClass('feature-selector')) {
+        $customValueSelector.val('');
+        $definedValueSelector.val('');
+      }
+
+      // if feature is defined and custom value is not defined
+      var disableDefinedFeature = !('' !== $featureSelector.val() && '' === $customValueSelector.val());
+
+      // if defined feature is available but empty, disable it too
+      if (!disableDefinedFeature) {
+        // recalculate options and waiting for ajax call to be done
+        var selector = '#' + $definedValueSelector.attr('id');
+        setTimeout(function() {
+          var nbOptions = document.querySelector(selector).options.length;
+          if (0 === nbOptions) {
+            disableDefinedFeature = true;
+          }
+        }, 500);
+      }
+
+      // if a defined value is set up, custom value is disabled
+      var disableValueFeature = $definedValueSelector.val() !== '';
+
+      console.log(disableDefinedFeature);
+      console.log(disableValueFeature);
+
+      $definedValueSelector.prop('disabled', disableDefinedFeature);
+      $customValueSelector.prop('disabled', disableValueFeature);
```

Comment:
To prevent the use of setTimeout, can this event handler be moved to the success of the AJAX call mentioned above?